SELinux guide

areed opened the pull request.

joshk commented:
This is awesome @areed 

@marccampbell could you give this a read too?

Also, should we add a summary of recommendations?

areed commented:
I added a summary and removed the section on customizing the Replicated type through the install scripts. I don't think we need to have a feature in the install scripts to change the domain now that we're using spc_t.

joshk commented on content/docs/kb/supporting-your-customers/selinux.md:
```diff
@@ -0,0 +1,100 @@
+volumes:
+- host_path: /dbdata
+  container_path: /var/lib/mysql
+  options: ["Z"]
```
should we consider using this as the default?

areed replied:
That could possibly lead to unexpected cases where the relabeling makes the directory inaccessible to other processes.

joshk replied:
🤔 

It feels like this should be the default, otherwise so many vendor applications won't work on SELinux by default.

To put it a different away, how can we apply some defaults so that things work by default?

I think at the very least we should update our example apps to work with SELinux by default.

We should also use a different example application (possibly using a gist), and then showing how to fix the issue.

joshk commented:
@areed this documentation is amazing! Thank you!

joshk commented:
@areed although I just merged this, could you add a note at the top about how this was addressed in version `2.12` of Replicated?
